packages/core/src/state-management/repository.ts

```diff
@@ -219,21 +219,8 @@ export class Repository {
     opts: CreateOpts | UpdateOpts
   ): Promise<RunningState> {
     const state$ = await this.stateManager.applyCommit(streamId, commit, opts)
-    await this.applyWriteOpts(state$, opts)
-    return state$
-  }
-
-  /**
-   * Apply options relating to authoring a new commit
-   *
-   * @param state$ - Running State
-   * @param opts - Initialization options (request anchor, publish to pubsub, etc.)
-   * @private
-   */
-  async applyWriteOpts(state$: RunningState, opts: CreateOpts | UpdateOpts) {
     this.stateManager.applyWriteOpts(state$, opts)
-
-    await this.handlePinOpts(state$, opts)
```

Comment:
So where are the pinning opts being handled now?

Reply:
Copied from https://github.com/ceramicnetwork/js-ceramic/pull/1862#discussion_r756456702:

@ukstv So i did some investigating and this is what I found:
`stream.update()`  --> `ceramic.applyCommit()` --> `repo.applyCommit()` 
It then calls
`stateManager.applyCommit()` --> `repo.load()` --> `repo.handlePinOpts()` --> `repo.unpin()`
After the above completes the following gets called 
`repo.applyWriteOpts()` --> `repo.handlePinOpts()` --> `repo.unpin()`

So essentially `repo.unpin()` gets called twice. This functions is async and so this double call may be whats confusing jest causing it to hang even after the test successfully completes . If I comment out one of `repo.handlePinOpts` calls the test passes and doesn't hang. If i don't unpin in the test, the test passes and doesn't hang.

I'm not sure if I am right so please help me! The duplicated call to unpin is unexpected 

---

Pinning is handled on `Repository::load` which is called from both `Repository::load` (obviously) and `Repository::applyCommit` (through `StateManager::applyCommit`).